### DISCUSSION TITLE
Slowness in MagickImage Object Creation

### DISCUSSION THREAD

**Discussion by MuteebAhmad148** [dlemstra/Magick.NET#1055] [Help]
Hi!
I'm trying to create a label Image and place it on top of another MagickImage.
![image](https://user-images.githubusercontent.com/54740149/137482236-f7087cc5-faf0-44c3-9231-9be845cf6aff.png)
![image](https://user-images.githubusercontent.com/54740149/137482614-0499a2cd-006d-428c-885d-cd3f59f1d76d.png)

But the issue is MagickImage object creation for label image is taking a long time (around 5 mints). Also, I'm using a long caption here which is around 2000 characters . It's works fine (still not that fast) with shorter captions.   

Thanks
Regards 
Muteeb Ahmad

**Comment by dlemstra** [comment]
Is it possible to provide a code sample that can be used to reproduce this issue?

**Comment by MuteebAhmad148** [comment]
Hi @dlemstra 
Thanks for your response.
I have attached the sample code file. You can use the attached background image to test the issue.
MagickImage Object creation is Line Number 36 is taking a long time to execute.

[Program.txt](https://github.com/dlemstra/Magick.NET/files/7357928/Program.txt)
![background](https://user-images.githubusercontent.com/54740149/137596118-36a924e2-4323-4556-9b64-2f1b9645f348.jpg)

**Comment by dlemstra** [comment]
We made some improvements to the caption coder of ImageMagick. And it might help if you use this new feature:

```C#
questionTextSettings.SetDefines(new CaptionReadDefines
{
    StartFontPointsize = 7, // You might need to tweak this value.
});
```

You should see an improvement but we will check if we can make some more improvements in the future.

**Comment by MuteebAhmad148** [comment]
Hi
Thanks for sharing this update.
I have tested it with latest version 8.4.0. It is still taking around 1 minute to create MagickImage object for 1000 character's caption, which is still a lot of time.

**Reply by jdmsolarius** [reply]
Your best Option for right now is probably to convert it to a Bitmap, and using System.Graphics.DrawString to create the Caption. the conversion will take less than 50 milliseconds, I know that's not Ideal but it will solve the problem

**Comment by dlemstra** [comment]
This is happening because the part that tries to figure out how to write the text without the specified boundaries is not that fast. But I haven't had time yet to see if we could speed up that part.
